src/components/chip/chip.tsx

```diff
@@ -178,10 +184,34 @@ export class Chip
     this.closed = true;
   };
 
+  private updateHasContent() {
+    const slottedContent = this.el.textContent.trim().length > 0 || this.el.childNodes.length > 0;
+    this.hasContent =
+      (getSlotted(this.el, SLOTS.image) || this.el.childNodes.length === 1) &&
```

Comment:
`getSlotted()` internal util is deprecated. If we put this in, can we create an issue to replace it with `onSlotChange`? the `onSlotChange` can set `hasContent` depending on how many assigned elements it has.

Reply:
I'll update to use `onSlotChange` and open a separate issue to track / remove places where `getSlotted` is currently used.

Reply:
https://github.com/Esri/calcite-components/issues/6059